src/mmda/predictors/hf_predictors/mention_predictor.py

```diff
@@ -91,12 +91,24 @@ def predict_page(self, page: Annotation, counter: Iterator[int], print_warnings:
             padding='max_length',
             return_overflowing_tokens=True,
             return_tensors="pt"
-        ).to(self.model.device)
-        del inputs["overflow_to_sample_mapping"]
+        )
 
         with torch.no_grad():
-            outputs = self.model(**inputs)
-        prediction_label_ids = torch.argmax(outputs.logits, dim=-1)
+            # Control device memory use to predictable levels
+            # by limiting size of batches sent to it.
+            prediction_label_ids = []
+            for index, sequence in enumerate(inputs["input_ids"]):
```

Comment:
Why not use a larger batch size here?

Reply:
The small batch size had no impact on inference speed, but reduces the max footprint of any one copy of the model. This means we can jam more copies onto one device, and shoudl be able to reap greater concurrency as a result.